### PR TITLE
Makefiles refactoring

### PR DESCRIPTION
**Pull request by gdoffe**
Big makefiles refactoring to clean and be more consistent on our working way.
This follow RIOT guidelines which gives us more flexibility by turning all software components into modules.
Note: a bug has been spotted in RIOT, I did the fix, thus our own RIOT must be used for now.

### CODE REVIEW

**Review by gdoffe — COMMENTED** [review]
Please say why.

**Review by gdoffe — COMMENTED** [review]
I love this callback, however say why you add it.
Here you just block automatic path pose increment on pose reached and you start the planner.
You do not initialize it.

Commit title always starts by a verb:
planners: shell: add a menu enter callback

File: platforms/cortex/platform.c [hunk 1]
```diff
@@ -265,18 +266,62 @@ static void *pf_task_countdown(void *arg)
...
+        printf("%d seconds left...\n", countdown);
+        countdown--;
+        xtimer_periodic_wakeup(&loop_start_time, US_PER_SEC);
+    }
+
+    /* Stop task_planner_start_cancel thread if still running */
+    thread_t *planner_start_cancel_thread = (thread_t *)thread_get(planner_start_cancel_pid);
+    if (planner_start_cancel_thread) {
+        sched_set_status(planner_start_cancel_thread, STATUS_STOPPED);
+    }
```
Comment (gdoffe) [inline platforms/cortex/platform.c:324]: Just a comment for later but I think this should be placed in a dedicated module.

File: planners/astar/planner.c [hunk 2]
```diff
@@ -78,11 +78,12 @@ static int trajectory_get_route_update(ctrl_t* ctrl, const pose_t* robot_pose,
             /* If the targeted pose is the current path pose to reach, launch
              * the action and update the current path pose to reach to the next
              * one in path, if allowed. */
-            DEBUG("planner: Controller has reach final position.\n");
+            DEBUG("planner: Controller has reached final position.\n");
 
-            /* If it exists launch action associated to the point and increment
-             * the position to reach in the path */
+            /* If not in calibration mode */
```
Comment (gdoffe) [inline planners/astar/planner.c:83]: shell mode ? (check commit message too)

File: planners/astar/planner.c [hunk 3]
```diff
@@ -93,14 +94,19 @@ static int trajectory_get_route_update(ctrl_t* ctrl, const pose_t* robot_pose,
             }
             /* Update current path targeted position in case it has changed */
             current_path_pos = path_get_current_path_pos(path);
+
             /* As current path pose could have changed, avoidance graph needs
-             * to be recomputed */
+                * to be recomputed */
```
Comment (gdoffe) [inline planners/astar/planner.c]: not aligned

File: planners/astar/planner.c [hunk 4]
```diff
@@ -93,14 +94,19 @@ static int trajectory_get_route_update(ctrl_t* ctrl, const pose_t* robot_pose,
...
+            avoidance_update = TRUE;
+        }
+        /* If in calibration mode and new pose is selected */
```
Comment (gdoffe) [inline planners/astar/planner.c:102]: shell

File: planners/astar/planner.c [hunk 5]
```diff
@@ -93,14 +94,19 @@ static int trajectory_get_route_update(ctrl_t* ctrl, const pose_t* robot_pose,
...
+        else if ((!allow_change_path_pose) && (ctrl_is_pose_intermediate(ctrl))) {
```
Comment (gdoffe) [inline planners/astar/planner.c:103]: if we have reach an intermediate position it should be `avoidance_index++;`, not `avoidance_update = TRUE;`.

Thats why I deleted it in a previous commit. What do you think ?

**Review by gdoffe — COMMENTED** [review]
Forgot calibration, it is shell now :P

**Review by gdoffe — COMMENTED** [review]
Dont you think we should activate uncrustify in the CI now ?
It would avoid this kind of huge commit in the future. ;)

File: platforms/cortex/platform.c (hunk 1 above)
Comment (ecourtois) [reply platforms/cortex/platform.c:324]: I agree and I thought about that too, it could be generic, maybe later.

**Comment by ecourtois** [comment]
Yes we should activate uncrustify in the CI.

**Comment by gdoffe** [comment]
> Yes we should activate uncrustify in the CI.

@Kheprys could you do that please ? maybe in a new branch after merge.

File: planners/astar/planner.c (hunk 5 above)
Comment (ecourtois) [reply planners/astar/planner.c:103]: I've just restored the former code that was working, but I don't clearly understand the planner behavior...

File: planners/astar/planner.c (hunk 2 above)
Comment (ecourtois) [reply planners/astar/planner.c:83]: Done

File: planners/astar/planner.c (hunk 4 above)
Comment (ecourtois) [reply planners/astar/planner.c:102]: Done

File: planners/astar/planner.c (hunk 3 above)
Comment (ecourtois) [reply planners/astar/planner.c]: The diff is weird, but the code is aligned...

**Comment by ecourtois** [comment]
Done, restored it from a previous commit!

**Comment by ecourtois** [comment]
`make check-codingrules` is already executed in CI, but result is not check, and it should be now.

**Comment by gdoffe** [comment]
LGTM, just the planner patch I sent you on discord to test. If it works fine for you too. you can merge.

File: lib/path/include/path.h [hunk 6]
```diff
@@ -6,7 +6,7 @@
 typedef struct {
     pose_t pos;
     uint8_t allow_reverse;
-    uint8_t max_speed;
+    double max_speed;
```
Comment (gdoffe) [inline lib/path/include/path.h:9]: You forgot to say why you did this in the commit message.

**Pull request merged**